store/store.go

```diff
@@ -746,6 +761,54 @@ func (s *Store[H]) deinit() {
 	s.heightSub.SetHeight(0)
 }
 
+// withWriteBatch attaches a new batch to the given context and returns cleanup func.
+func (s *Store[H]) withWriteBatch(ctx context.Context) (context.Context, func() error) {
+	bds, ok := s.ds.Children()[0].(datastore.Batching)
+	if !ok {
+		return ctx, func() error { return nil }
+	}
+
+	if _, ok = contextds.GetWrite(ctx); ok {
```

Comment:
Is it concurrency safe?  Seems like two batches can be created in parallel. Just to make sure it is not possible that 2 batches are created in parallel any any batch is lost